### PR TITLE
No issue: Use correct server host and force subscribe for push.

### PR DESCRIPTION
**Pull request by jonalmeida**
This should enable push notifications to work for new installs and existing nightly ones.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by jonalmeida** [comment]
@grigoryk should this be blocked on https://github.com/mozilla-mobile/android-components/issues/3827?

File: app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt [hunk 1]
```diff
@@ -101,28 +101,29 @@ class BackgroundServices(
...
+                                    authKey = subscription.authKey
+                                )
                             )
-                        )
+                    }
                 }
             }, ProcessLifecycleOwner.get(), false)
+
+            // We need to force subscribe for nightly users before push server support was added.
+            it.subscribeForType(PushType.Services)
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt]: Looks like you'll run this on every startup. That doesn't sound right?

File: app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt [hunk 2]
```diff
@@ -101,28 +101,29 @@ class BackgroundServices(
...
-                    accountManager.authenticatedAccount()?.deviceConstellation()
-                        ?.setDevicePushSubscriptionAsync(
-                            DevicePushSubscription(
-                                endpoint = subscription.endpoint,
-                                publicKey = subscription.publicKey,
-                                authKey = subscription.authKey
+                    // Update for only the services subscription.
+                    if (subscription.type == PushType.Services) {
+                        accountManager.authenticatedAccount()?.deviceConstellation()
+                            ?.setDevicePushSubscriptionAsync(
```
Comment (grigoryk) [inline app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt]: The `subscribeForType` call below will result in `onSubscriptionAvailable` being called. So you'd call this method on every startup. Underneath, it's expected to modify an FxA device record. Seems like you'll write to FxA servers on every app startup here, which isn't something we can land as-is.

File: app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt (hunk 1 above)
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt]: If we don't have a valid subscription, we hit the server, if we return it locally (this is all done in the rust side).

**Comment by jonalmeida** [comment]
Updated PR and added https://github.com/mozilla-mobile/android-components/pull/3849 to fix the concerns.

File: app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt (hunk 2 above)
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/components/BackgroundServices.kt]: I've updated the PR to remove the token from prefs and force registration only once. We can remove this in https://github.com/mozilla-mobile/fenix/issues/4218 when majority of our users are on the latest nightly or release.

**Comment by jonalmeida** [comment]
Updated the PR based on newer recent discussions and dependant a-c changes.

**Comment by jonalmeida** [comment]
> Updated PR and added [mozilla-mobile/android-components#3849](https://github.com/mozilla-mobile/android-components/pull/3849) to fix the concerns.

No longer blocked! ✨

**Pull request merged**